src/app/components/TaskCard.tsx

```diff
@@ -11,48 +11,53 @@ type TaskCardProps = {
   return (
-    <div
-      className={`mt-6 rounded-xl flex justify-between py-4 px-2 ${
-        task.isCompleted ? "bg-[rgba(13, 153, 255, 0.1)]" : "bg-white"
-      }`}
-    >
-      <div className="flex flex-col gap-2">
-        <div className="flex gap-2 items-center">
-          <span className="w-4 h-4 rounded-full bg-black" />
-          <span className="text-base">{task.value}</span>
-        </div>
-        <div className="flex gap-2 items-center">
-          <CalendarIcon />
-          <span className="text-sm text-gray-600">Due Date: </span>
-          <span className="text-sm">
-            {task.dueDate ? task.dueDate : "No date"},{" "}
-            {task.dueTime ? task.dueTime : "No time"}
-          </span>
-        </div>
-        <PriorityComplexityDisplay priority={task.priority} />
-        <PriorityComplexityDisplay complexity={task.complexity} />
-        {task.tags.length > 0 && (
-          <div className="mt-1 flex gap-2 flex-wrap">
-            {task.tags.map((tag) => (
-              <span key={tag.id} className="text-xs bg-blue-100 p-1 rounded-lg">
-                {tag.value}
-              </span>
-            ))}
+    <Link href={`/task/${task.id}`}>
+      <div
+        className={`mt-6 rounded-xl flex justify-between py-4 px-2 ${
+          task.isCompleted ? "bg-[rgba(13, 153, 255, 0.1)]" : "bg-white"
+        }`}
+      >
+        <div className="flex flex-col gap-2">
+          <div className="flex gap-2 items-center">
+            <span className="w-4 h-4 rounded-full bg-black" />
+            <span className="text-base">{task.value}</span>
           </div>
-        )}
-      </div>
-      <div className="flex flex-col">
-        <div className="flex gap-4">
-          <Link href={`task/edit/${task.id}`}>
-            <button className="bg-blue-100 rounded-full p-2 w-8 h-8">
-              <EditIcon />
+          <div className="flex gap-2 items-center">
+            <CalendarIcon />
+            <span className="text-sm text-gray-600">Due Date: </span>
+            <span className="text-sm">
+              {task.dueDate ? task.dueDate : "No date"},{" "}
+              {task.dueTime ? task.dueTime : "No time"}
+            </span>
+          </div>
+          <PriorityComplexityDisplay priority={task.priority} />
+          <PriorityComplexityDisplay complexity={task.complexity} />
+          {task.tags.length > 0 && (
+            <div className="mt-1 flex gap-2 flex-wrap">
+              {task.tags.map((tag) => (
+                <span
+                  key={tag.id}
+                  className="text-xs bg-blue-100 p-1 rounded-lg"
+                >
+                  {tag.value}
+                </span>
+              ))}
+            </div>
+          )}
+        </div>
+        <div className="flex flex-col">
+          <div className="flex gap-4">
+            <Link href={`task/edit/${task.id}`}>
+              <button className="bg-blue-100 rounded-full p-2 w-8 h-8">
```

Comment:
We should create a reusable button because this button and the button which have CompleteIcon use same styles.